wikimedia/wikipedia-ios
# Improve usability of search languages bar (1 of 2)

**Phabricator:** https://phabricator.wikimedia.org/T277058

### Notes
This is part 1 of 2 in improving the usability of the language bar in the search controller. The specific task elements addressed by this PR are:
1. Allow all of user's preferred languages to be displayed in search language bar (instead of the previous limitation of 3)
2. Increase the size of the "more languages" gradient overlay to match design
3. Updates "More" button title color to persistently be the theme's primary link color. 
4. Update displayed language name to match its display in the preferred "Wikipedia languages" view controller. That is, the name of a language in the search language bar should match that language's displayed header text in the "Wikipedia languages" view controller.

The next PR will address the design updates.

### Test Steps
Confirm there is behavioral parity with the existing search bar, excepting the changes noted above. Some examples include:
1. Confirm adding and removing languages results in the proper number of language bar buttons displayed.
2. Confirm tapping a search bar button and making a search functions as it did before.
3. Confirm it renders as expected in all themes.

### Screenshot
<img width="889" alt="before_after_language_bar" src="https://user-images.githubusercontent.com/5652327/113619355-b7f59b80-960d-11eb-84fd-9ea24ef70975.png">
